### ISSUE TITLE
Exclude-Pattern for a specifc sniff don't seem to work

### ISSUE THREAD

**Issue by finn-matti** [wongjn/vscode-php-sniffer#67]
**Describe the bug**
An exclude pattern for a specific sniff like so:

```xml
<!-- Do not warn about _before/_after in codeception tests (relevant for IDEs) -->
<rule ref="PSR2.Methods.MethodDeclaration.Underscore">
    <exclude-pattern>*/tests/*</exclude-pattern>
</rule>
```

And a codeception test like this:

```php
<?php

class MinimalCest
{

    public function _before(UnitTester $I)
    {
    }

    // tests
    public function minimalCheck(UnitTester $I)
    {
        $I->assertTrue(true);
    }
}

```

Doesn't seem to be working correctly.

**To Reproduce**
1. Include the snippet above in your phpcs.xml
2. You'll still see the error about _before/_after in codeception cest files

![Screenshot 2023-05-03 at 12 29 13](https://user-images.githubusercontent.com/56148301/235880289-eaa581fa-7f80-4946-ba88-f292d2079530.png)

**Expected behavior**
Just as phpcs run from the commandline, the exclude pattern should be honored:

Without the snippet from above:

![Screenshot 2023-05-03 at 12 30 09](https://user-images.githubusercontent.com/56148301/235880440-8960f1fe-5ff2-496d-af93-bad3c6eea4b1.png)

With the snippet from above:

![Screenshot 2023-05-03 at 12 30 41](https://user-images.githubusercontent.com/56148301/235880544-23b589aa-e256-4c0f-9bfd-78866b65ff5d.png)

**Environment**
 - OS: MacOs Ventura 13.3.1
 - VSCode version: 1.77.3
 - PHP CLI version: 8.1.18
 - squizlabs/PHP_CodeSniffer version: 1.3.0 (global ✔)

**Extension settings**
```json
{
    "phpSniffer.run": "onSave",
    "phpSniffer.onTypeDelay": 250,
    "phpSniffer.executablesFolder": "",
    "phpSniffer.autoDetect": false,
    "phpSniffer.standard": "config/phpcs.xml",
    "phpSniffer.snippetExcludeSniffs" : [],
}
```

**Comment by wongjn** [comment]
Unfortunately, I'm unable to reproduce this in a test project with the given details. Consider checking whether `phpSniffer.standard` path is correct – change the content of `config/phpcs.xml` and see if the extension picks it up (reload the window to make sure).

**Comment by finn-matti** [comment]
@wongjn Damn. Is there a way to put the extension in debug mode? To see what it sends to phpcs? I can't for the life of me see a difference in the paths, but your intuition was right: My config is not being picked up, it seems.

**Comment by wongjn** [comment]
There is no debug mode. In case it helps, the `standard` path is relative to the workspace folder.

**Comment by finn-matti** [comment]
AHA!

I managed to hack around in the plugin (located at "~/.vscode/extensions/wongjn.php-sniffer-1.3.0") a little and can now confirm that the wrong standard is handed over:

I added something in `/lib/runner.js` after requiering window from vscode above:

```js
if (standard) args.set('standard', standard);

window.showInformationMessage(standard);
```

![Screenshot 2023-05-03 at 18 47 00](https://user-images.githubusercontent.com/56148301/235968780-f3bf44bb-e4dd-45cc-8b48-da530950a672.png)

Pretty interesting!

Here is what my user settings for the extension look like:

```json
"phpSniffer.run": "onSave",
"phpSniffer.onTypeDelay": 250,
"phpSniffer.executablesFolder": "",
"phpSniffer.autoDetect": false,
"phpSniffer.standard": "",
"phpSniffer.snippetExcludeSniffs": []
```

Here is what my workspace settings for the extension look like:

```json
"phpSniffer.run": "onSave",
"phpSniffer.onTypeDelay": 250,
"phpSniffer.executablesFolder": "",
"phpSniffer.autoDetect": false,
"phpSniffer.standard": "config/phpcs.xml",
"phpSniffer.snippetExcludeSniffs": []
```

If anything I would have expected the PEAR standard, because that is [the default](https://github.com/squizlabs/PHP_CodeSniffer#getting-started) for phpcs.

Could there be a caching issue?

**Comment by wongjn** [comment]
I don't think there would be any caching there. It would not be from PHP_CodeSniffer since `standard` is taken direct from configuration: https://github.com/wongjn/vscode-php-sniffer/blob/2305dc6be1c683e2a65fdb1e54bf8356421c6e7e/lib/runner.js#L105

It seems like you're mistaken in what the configuration is for the project and the `phpSniffer.standard` value is `PSR12`.

**Comment by finn-matti** [comment]
You see the configs above yourself. I opened the json files through vscode's settings browser:

![Screenshot 2023-05-03 at 19 15 41](https://user-images.githubusercontent.com/56148301/235976920-8b94a572-6e66-4614-9b48-44e849a9b754.png)

I think you're right that it's not this extension's fault, but a little weird none the less. If it's okay with you I'd still like to leave  the issue open for the moment so I can close this issue with a solution, if other folks run into this ever.

EDIT: It doesn't matter if it's open or closed, I can still comment. Issue can terefore be close. 👍

**Comment by finn-matti** [comment]
Found it!

I had a `.vscode/settings.json` with the following contents:

```json
{
    "phpSniffer.standard": "PSR12",
}
```

in my repo.

May it help others in the future!
